python/lsst/ap/association/assoc_db_sqlite.py

```diff
@@ -0,0 +1,479 @@
+
+import sqlite3
+
+from lsst.meas.algorithms.indexerRegistry import IndexerRegistry
+import lsst.afw.table as afwTable
+import lsst.afw.geom as afwGeom
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+from .dia_collection import *
+from .dia_object import *
+
+
+__all__ = ["AssociationDBSqliteConfig", "AssociationDBSqliteTask"]
+
+
+afw_to_db_types = {
+    "L": "INTEGER",
+    "Angle": "REAL"
+}
+
+
+class AssociationDBSqliteConfig(pexConfig.Config):
+    """!
+    \anchor AssociationDBSqliteConfig_
+
+    \brief Configuration parameters for the AssociationDBSqliteTask
+    """
+    db_name = pexConfig.Field(
+        dtype=str,
+        doc='Location on disk and name of the sqlite3 database for storing '
+        'and loading DIASources and DIAObjects.',
+        default=':memory:'
+    )
+    indexer = IndexerRegistry.makeField(
+        doc='Select the spatial indexer to use within the database.',
+        default='HTM'
+    )
+
+
+class AssociationDBSqliteTask(pipeBase.Task):
+    """!
+    \anchor AssociationDBSqliteTask_
+
+    \brief Enable storage of reading of DIAObjects and DIASources from a
+    sqlite database.
+
+    \section ap_association_AssociationDBSqliteTask_Contents Contents
+      - \ref ap_association_AssociationDBSqliteTask_Purpose
+      - \ref ap_association_AssociationDBSqliteTask_Config
+      - \ref ap_association_AssociationDBSqliteTask_Run
+      - \ref ap_association_AssociationDBSqliteTask_Debug
+      - \ref ap_association_AssociationDBSqliteTask_Example
+
+    \section ap_association_AssociationDBSqliteTask_Purpose   Description
+    Create a simple sqlite database and implement wrappers to store and
+    retrieve DIAObjects and DIASources within that database. This task
+    functions as a testing ground for the L1 database and should mimic this
+    database's eventual functionaly. This specific database implementation is
+    useful for the verification packages which may not be run with access to
+    L1 database.
+
+    \section ap_association_AssociationDBSqliteTaskInitialize
+        Task initialization
+    \copydoc \_\_init\_\_
+
+    \section ap_association_AssociationDBSqliteTask_Run
+        Invoking the Task
+    \copydoc run
+
+    \section ap_association_AssociationDBSqliteTask_Config
+        Configuration parameters
+    See \ref AssociationDBSqliteConfig
+
+    \section ap_association_AssociationDBSqliteTask_Debug
+        Debug variables
+    This task has no debug variables
+
+    \section ap_association_AssociationDBSqliteTask_Example
+        Example of using AssociationDBSqliteTask
+    This task has no standalone example, however it is applied as a subtask of
+    ap.association.AssociationTask
+    """
+
+    ConfigClass = AssociationDBSqliteConfig
+    _DefaultName = "association_db_sqlite"
+
+    def __init__(self, **kwargs):
+        """ Create a new connection to the sqlite database specified in
+        the config.
+        """
```

Comment:
Not sure about the arguments. This set up seems to be pretty standard for other Tasks I've seen in the stack. All of the variables are taken care of in the config and anything sent in **kwargs would go to the base class __init__ if needed. At least that's my understanding of why things are this way.

Reply:
Hmm... fine, I guess if it is [standard](https://pipelines.lsst.io/v/DM-11253/modules/lsst.pipe.base/creating-a-task.html#introduction) documenting it would be redundant.  :confused: